src/device/device_ledger.cpp

```diff
@@ -185,41 +185,8 @@ namespace hw {
     #define INS_GET_RESPONSE                    0xc0
 
 
-    void device_ledger::logCMD() {
-      if (apdu_verbose) {
-        char  strbuffer[1024];
-        snprintf(strbuffer, sizeof(strbuffer), "%.02x %.02x %.02x %.02x %.02x ",
-          this->buffer_send[0],
-          this->buffer_send[1],
-          this->buffer_send[2],
-          this->buffer_send[3],
-          this->buffer_send[4]
-          );
-        const size_t len = strlen(strbuffer);
-        buffer_to_str(strbuffer+len, sizeof(strbuffer)-len, (char*)(this->buffer_send+5), this->length_send-5);
-        MDEBUG( "CMD  :" << strbuffer);
-      }
-    }
-
-    void device_ledger::logRESP() {
-      if (apdu_verbose) {
-        char  strbuffer[1024];
-        snprintf(strbuffer, sizeof(strbuffer), "%.02x%.02x ",
-          this->buffer_recv[this->length_recv-2],
-          this->buffer_recv[this->length_recv-1]
-          );
-        const size_t len = strlen(strbuffer);
-        buffer_to_str(strbuffer+len, sizeof(strbuffer)-len, (char*)(this->buffer_recv), this->length_recv-2);
-        MDEBUG( "RESP :" << strbuffer);
-
-      }
-    }
-
-    /* -------------------------------------------------------------- */
-    device_ledger::device_ledger() {
+    device_ledger::device_ledger(): hw_device(0x0101, 0x05, 64, 10000) {
```

Comment:
please construct ```hw_device``` explicitly so some else (me) can instantiate their own implementation of ```hw::io::device_io``` (with ```#ifdef```s)

Reply:
do you suggest to declare:

    hw::io::device_io *hw_device;

and do a:

    hw_device = new hw::io::device_io_hid((0x0101, 0x05, 64, 10000);

or do you suggest the contructor take the pointer as parameter? 
how to you plan to use that? I'm not sure it solve your pb
It is not clear for me what I zm suppose to mod :)

Reply:
sorry, i was over-thinking it - leave it like it is.

Reply:
hmm ...

Ok So I will 
- leave as is for know,
- make some cleanup, 
- squash all 
- remove the "donot merge"

:)